File: Lab 10/t3.c
#include <stdio.h>

int t (const int n)
{
  if (n == 1)
    return 1;
  return t (n / 2) + n;
}

int main (void)
{
  int input = 0;
  int r = 0;

  puts ("Please input a positive integer:");
  r = scanf ("%d", &input);
  if (r != 1) {
    puts ("Illegal input.");
    return 1;
  }

  printf ("T(%d)=%d\n", input, t (input));

  return 0;
}

File: Lab 10/t3.go
package main

import (
  "fmt"
  "os"
  "bufio"
)

func main() {
  var (
    input int
    e error
  )

  stdio := bufio.NewReader(os.Stdin)

  fmt.Println("Please input a positive integer:")
  _, e = fmt.Fscanf(stdio, "%d", &input)
  if e != nil {
    fmt.Fprintln(os.Stderr, "Illegal input:", e)
    return
  }

  fmt.Printf("T(%d)=%d\n", input, t(input))
}

func t(n int) int {
  if n == 1 {
    return 1
  }
  return t(n/2) + n
}
